src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -153,6 +153,10 @@ public function load(array $configs, ContainerBuilder $container)
             $definition->replaceArgument(2, E_COMPILE_ERROR | E_PARSE | E_ERROR | E_CORE_ERROR);
         }
 
+        if (interface_exists('Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface')) {
```

Comment:
there should also be a config setting to force disabling it in case you don't want to use the parameter converter, allowing to get rid of its overhead